src/Libs/Gio-2.0/Public/Application.cs

```diff
@@ -14,19 +14,21 @@ public int Run()
         return Internal.Application.Run(Handle, 0, new string[0]);
     }
 
-    public int RunWithSynchronizationContext()
+    public int RunWithSynchronizationContext(int argc, string[] argv)
     {
         var original = SynchronizationContext.Current;
 
         SynchronizationContext.SetSynchronizationContext(new GLib.Internal.MainLoopSynchronizationContext());
 
         try
         {
-            return Run();
+            return Run(argc, argv);
         }
         finally
         {
             SynchronizationContext.SetSynchronizationContext(original);
         }
     }
+
+    public int RunWithSynchronizationContext() => RunWithSynchronizationContext(0, new string[0]);
```

Comment:
`Array.Empty<string>()` would be better than `new string[0]`

Reply:
@badcel should this be changed on line 14 as well?